feat(snapcraft_legacy): pack command runs the lifecycle (#5616)
snapcore/snapcraft: docs/release-notes/snapcraft-8-11.rst

See :ref:`reference-system-requirements` for information on the minimum hardware and
installed software.

What's new
----------

Snapcraft 8.11 brings the following features, integrations, and improvements.

Unified behavior for the pack command
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

When a new base is introduced in Snapcraft, major breaking changes are made only for the
new base. This allows Snapcraft to introduce changes without breaking builds of existing

> snaps on older bases.

Author: https://github.com/canonical/snapcraft/pull/5616/commits/52a9e1e69869973a0dec624ac9882551b05a7a59

> This strategy has its downsides. For example, the user experience can be inconsistent
> when building snaps with different bases. ``snapcraft pack`` would behave differently
> depending on the base.

Reviewer: ```suggestion
This strategy has downsides. In particular, the user experience can be inconsistent when
building snaps with different bases. Running ``snapcraft pack`` would behave differently
depending on the base.
```

> Now, the pack command for core20 snaps has been extended to have the same functionality
> as the pack command for core22 and core24 snaps. With no arguments, Snapcraft builds and
> packages a snap. If a directory is provided, Snapcraft will skip the build process and
> instead pack the directory as a snap.

Reviewer: ```suggestion
With this release, the pack command for core20 snaps is ported to have the same
functionality as the pack command for core22 and core24 snaps. With no arguments,
running ``snapcraft`` builds and packages the snap. If a directory is passed as the
first positional argument, Snapcraft skips the build process and instead pack the
directory as a snap.
```

Author: https://github.com/canonical/snapcraft/pull/5616/commits/52a9e1e69869973a0dec624ac9882551b05a7a59
